app/controllers/YourSelfAssessmentChargesController.scala

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2025 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import audit.AuditingService
+import audit.models.WhatYouOweResponseAuditModel
+import auth.MtdItUser
+import auth.authV2.AuthActions
+import config.featureswitch.FeatureSwitching
+import config.{AgentItvcErrorHandler, FrontendAppConfig, ItvcErrorHandler, ShowInternalServerError}
+import enums.GatewayPage.WhatYouOwePage
+import forms.utils.SessionKeys.gatewayPage
+import models.admin.{AdjustPaymentsOnAccount, CreditsRefundsRepay, FilterCodedOutPoas, ReviewAndReconcilePoa, YourSelfAssessmentCharges}
+import models.core.Nino
+import models.nextPayments.viewmodels.WYOClaimToAdjustViewModel
+import play.api.Logger
+import play.api.i18n.{I18nSupport, Messages}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents, Result}
+import services.{ClaimToAdjustService, DateServiceInterface, WhatYouOweService}
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import views.html.WhatYouOwe
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class YourSelfAssessmentChargesController @Inject()(val authActions: AuthActions,
+                                     val whatYouOweService: WhatYouOweService,
+                                     val claimToAdjustService: ClaimToAdjustService,
+                                     val itvcErrorHandler: ItvcErrorHandler,
+                                     val itvcErrorHandlerAgent: AgentItvcErrorHandler,
+                                     val auditingService: AuditingService,
+                                     val dateService: DateServiceInterface,
+                                     whatYouOwe: WhatYouOwe
+                                    )(implicit val appConfig: FrontendAppConfig,
+                                      val mcc: MessagesControllerComponents,
+                                      val ec: ExecutionContext) extends FrontendController(mcc) with I18nSupport with FeatureSwitching{
+
+
+  def handleRequest(backUrl: String,
+                    itvcErrorHandler: ShowInternalServerError,
+                    isAgent: Boolean,
```

Comment:
as far as I understand `isAgent` flag is no longer required and you should be able to get this info from` user: MtdItUser[_]`